keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensor.java

```diff
@@ -151,22 +157,22 @@ public IntegerTensor pow(int exponent) {
 
     @Override
     public IntegerTensor minus(IntegerTensor that) {
-        return duplicate().minusInPlace(that);
+        return duplicate().minusInPlace(that.duplicate());
```

Comment:
Why have we added `duplicate` on these?

Reply:
Because we don't want the operation to be done in place, and we don't know which way around it will be applied at this point (as in, `this.minus(that)` or `that.minus(this)`).

Reply:
but then shouldn't the `duplicate()` happen inside `minusInPlace`?

Reply:
But we don't want the tensors duplicated if we just call `minusInPlace` directly, in that case we'd expect the operation to be done in place and the original arguments to be modified in the process. Correct me if I'm wrong though.

Reply:
Oh, I just read up on the difference between in-place and not in-place operations, and I think I see what your saying. With ND4J's `x.subi(y)` you'd expect `x` to be modified but `y` to remain the same. My bad, I had misunderstood it previously. I'll change this around.